File: python/string/generateTheString.py
#!/usr/bin/env python3
#coding=utf-8

''' leetcode issue 1374'''
import string

class Solution:
    def generateTheString(self, n: int) -> str:
        ret = 'a'
        while(n >= 26):
            n -= 2
            ret += "aa"
        ret += string.ascii_lowercase[1:n-1]
        return ret

File: c++/string/generateTheString.cc
#include <string>
#include <iostream>

using namespace std;

/* leetcode issue 1374 */
class Solution {
public:
    string generateTheString(int n) {
        string s = "abcdefghijklmnopqrstuvwxyz";
        string ret("a");

        while(n >= 26) {
            ret += "aa";
            n-= 2;
        }
        ret += s.substr(1, n-1);
        return ret;
    }
};

